pipelines/utils/secrets.py

```diff
@@ -1,10 +1,29 @@
 from google.cloud import secretmanager
 import json
 import pipelines.utils.env as env
+import requests
 
-DEFAULT_PROJECT_NUM = env.get("GCP_PROJECT_NUM")
+def get_project_id():
+  try:
+    response = requests.get(
+      'http://metadata.google.internal/computeMetadata/v1/project/project-id',
+      headers={'Metadata-Flavor': 'Google'},
+      timeout=3  # Add timeout to avoid hanging
+    )
+    response.raise_for_status()  # Raise exception for HTTP errors
+    return response.text
+  except (requests.RequestException, ConnectionError):
+    # Fall back to environment variable if metadata server request fails
+    import os
```

Comment:
Is `os` being used? If so, I find that it's easier to manage libraries and imports at the very top of the file. If not using, you can remove this import.